File: SoalB/Pyhton/PC.py
# memanggil class
from CPU import CPU
from RAM import RAM
from DISK import DISK

# atribut
__cpuA = CPU()
__ramB = RAM()
__diskC = DISK()
__pcPrice = 0


class PC():
    # constructor
    def __init__(self):
        self.__cpuA = CPU()
        self.__ramB = RAM()
        self.__diskC = DISK()
        self.__pcPrice = 0

    # membuat get set method class CPU
    def setCPU(self, cpuA):
        self.__cpuA = cpuA

    def getCPU(self):
        return self.__cpuA

    # membuat get set method class RAM
    def setRAM(self, ramB):
        self.__ramB = ramB

    def getRAM(self):
        return self.__ramB

    # membuat get set method class DISK
    def setDISK(self, diskC):
        self.__diskC = diskC

    def getDISK(self):
        return self.__diskC

    # membuat get set method total harga PC
    def setPcPrice(self, pcPrice):
        self.__pcPrice = pcPrice

    def getPcPrice(self):
        return self.__pcPrice

    # menampilkan data pc
    def printPC(self):
        self.__cpuA.printCPU()
        self.__ramB.printRAM()
        self.__diskC.printDISK()
        print("PC Price(IDR)        : " + str(self.getPcPrice()))
        print("---------------------------------------")

File: SoalA/C++/PC.cpp
#include <iostream>
#include <string>

using namespace std;

/* memanggil class */
#include "CPU.cpp"
#include "RAM.cpp"
#include "DISK.cpp"

class PC
{
    /* membuat atribut */
private:
    CPU cpuA;
    RAM ramB;
    DISK diskC;
    int pcPrice;

public:
    /* constructor */
    PC(CPU cpuA, RAM ramB, DISK diskC, int pcPrice)
    {
        this->cpuA = cpuA;
        this->ramB = ramB;
        this->diskC = diskC;
        this->pcPrice = pcPrice;
    }

    PC()
    {
    }

    /* Membuat get set untuk class CPU */
    void setCPU(CPU cpuA)
    {
        this->cpuA = cpuA;
    }
    CPU getCPU()
    {
        return this->cpuA;
    }

    /* Membuat get set untuk class RAM */
    void setRAM(RAM ramB)
    {
        this->ramB = ramB;
    }
    RAM getRAM()
    {
        return this->ramB;
    }

    /* Membuat get set untuk class DISK */
    void setDISK(DISK diskC)
    {
        this->diskC = diskC;
    }
    DISK getDISK()
    {
        return this->diskC;
    }

    /* Membuat get set untuk menghitung total harga */
    void setPcPrice(int pcPrice)
    {
        this->pcPrice = pcPrice;
    }
    int getPcPrice()
    {
        return this->pcPrice;
    }

    /* menampilkan data semua */
    void printPC()
    {
        this->cpuA.printCPU();
        this->ramB.printRAM();
        this->diskC.printDISK();
        cout << "PC Price(IDR)      : " << this->getPcPrice() << endl;
        cout << "====================================================" << endl;
    }

    /* destructor */
    ~PC()
    {
    }
};
